modules/api/pkg/handler/v2/routes_v2.go

```diff
@@ -1512,6 +1513,121 @@ func (r Routing) RegisterV2(mux *mux.Router, oidcKubeConfEndpoint bool, oidcCfg
 		Handler(r.getSeedOverview())
 }
 
+// swagger:route GET /api/v2/projects/{project_id}/disktypes project listProjectGCPDisk
+//
+//	Creates a cluster for the given project.
+//
+//	Consumes:
+//	- application/json
+//
+//	Produces:
+//	- application/json
+//
+//	Responses:
+//	  default: errorResponse
+//	  201: Cluster
+//	  401: empty
+//	  403: empty
+func (r Routing) listProjectGCPDiskTypes() http.Handler {
+	return httptransport.NewServer(
+		endpoint.Chain(
+			middleware.TokenVerifier(r.tokenVerifiers, r.userProvider),
+			middleware.UserSaver(r.userProvider),
+		)(provider.ListProjectGCPDiskTypes(r.presetProvider, r.userInfoGetter, true)),
+		provider.DecodeGCPSubnetworksNoCredentialReq,
+		handler.EncodeJSON,
+		r.defaultServerOptions()...,
+	)
+}
+
+// swagger:route GET /api/v2/projects/{project_id}/providers/gcp/subnetworks gcp listProjectGCPSubnetworks
+//
+// Lists available GCP subnetworks
+//
+//	Produces:
+//	- application/json
+//
+//	Responses:
+//	  default: errorResponse
+//	  200: GCPSubnetworkList
+func (r Routing) listProjectGCPSubnetworks() http.Handler {
+	return httptransport.NewServer(
+		endpoint.Chain(
+			middleware.TokenVerifier(r.tokenVerifiers, r.userProvider),
+			middleware.UserSaver(r.userProvider),
+		)(provider.ListProjectGCPSubnetworks(r.presetProvider, r.userInfoGetter, r.seedsGetter, true)),
+		provider.DecodeGCPSubnetworksNoCredentialReq,
```

Comment:
All `Req` structs need their own correct `Decode` function. This uses a Decode function that won't include project information, as an example.

Reply:
This applies to all of the `list...()` funcs btw. Here's an example of how one of the decode funcs look like in the Nutanix implementation: https://github.com/kubermatic/dashboard/blob/91bdd42fa485303eda47aa1fefbc9335a9502814/modules/api/pkg/handler/v2/provider/nutanix.go#L193-L210